lib/active_model/serializer/attributes.rb

```diff
@@ -1,55 +1,23 @@
 module ActiveModel
   class Serializer
     module Attributes
-      # @api private
-      class Attribute
-        delegate :call, to: :reader
-
-        attr_reader :name, :reader
-
-        def initialize(name)
-          @name = name
-          @reader = :no_reader
-        end
-
-        def self.build(name, block)
-          if block
-            AttributeBlock.new(name, block)
-          else
-            AttributeReader.new(name)
-          end
-        end
-      end
-      # @api private
-      class AttributeReader < Attribute
-        def initialize(name)
-          super(name)
-          @reader = ->(instance) { instance.read_attribute_for_serialization(name) }
-        end
-      end
-      # @api private
-      class AttributeBlock < Attribute
-        def initialize(name, block)
-          super(name)
-          @reader = ->(instance) { instance.instance_eval(&block) }
-        end
-      end
-
       extend ActiveSupport::Concern
 
       included do
         with_options instance_writer: false, instance_reader: false do |serializer|
-          serializer.class_attribute :_attribute_mappings # @api private : maps attribute key names to names to names of implementing methods, @see #attribute
-          self._attribute_mappings ||= {}
+          serializer.class_attribute :_attributes_data # @api private
+          self._attributes_data ||= {}
```

Comment:
@beauby For anyone looking back at this, would mind jotting down the advantage of the name change?

I know that part of it is that 
- in both cases the `Attribute` can get the serialization value given a block or record's attribute name, and then
- whether or not the `Attribute` should *also* know the serialization key used for its value/

```diff
-          serializer.class_attribute :_attribute_mappings # @api private : maps attribute key names to names to names of implementing methods, @see #attribute
+          serializer.class_attribute :_attributes_data # @api private
```

```diff
       def attributes(requested_attrs = nil, reload = false)
            @attributes = nil if reload
 -          @attributes ||= self.class._attribute_mappings.each_with_object({}) do |(key, attribute_mapping), hash|
 +          @attributes ||= self.class._attributes_data.values.each_with_object({}) do |attr, hash|
```

```diff
         def attribute(attr, options = {}, &block)
 -          _attribute_mappings[key] = Attribute.build(attr, block)
 +          _attributes_data[attr] = Attribute.new(attr, key, block)
```
```diff
-        # names of attribute methods
+        # keys of attributes
          # @see Serializer::attribute
          def _attributes
-          _attribute_mappings.keys
+          _attributes_data.values.map(&:key)
```

```diff
      def _attributes_keys
 -          _attribute_mappings
+          _attributes_data.values
 -            .each_with_object({}) do |(key, attribute_mapping), hash|
+            .each_with_object({}) do |attr, hash|
 -              next if key == attribute_mapping.name
+              next if attr.key == attr.name
-              hash[attribute_mapping.name] = { key: key }
+              hash[attr.name] = { key: attr.key }
```

```diff
+    Attribute = Struct.new(:name, :key, :block) do
-                                        (name, block)
```